src/common/vector/value_vector_utils.cpp

```diff
@@ -11,15 +11,24 @@ void ValueVectorUtils::copyNonNullDataWithSameTypeIntoPos(
     case STRUCT: {
         for (auto& childVector : StructVector::getChildrenVectors(&resultVector)) {
             copyNonNullDataWithSameTypeIntoPos(*childVector, pos, srcData);
-            srcData += childVector->getNumBytesPerValue();
+            srcData += Types::getDataTypeSize(childVector->dataType);
```

Comment:
open an issue about removes Types::getDataTypeSize because for different layout, a data type requires difference sizes.

Reply:
opened #1519